src/action/stat.rs

```diff
@@ -46,18 +46,28 @@ pub struct Response {
 
 #[derive(Debug)]
 pub struct Request {
-    pathspec: Option<PathSpec>,
-    collect_ext_attrs: Option<bool>,
-    follow_symlink: Option<bool>,
+
+    pathspec: PathSpec,
+    collect_ext_attrs: bool,
+    follow_symlink: bool,
+}
+
+#[derive(Debug)]
+pub struct ExtAttr {
+
+    name: Vec<u8>,
+    value: Vec<u8>,
 }
 
 #[derive(Debug)]
 struct PathSpec {
+
     nested_path: Option<Box<PathSpec>>,
     path: Option<PathBuf>,
 }
```

Comment:
Since you collapse the path as the first thing in you `handle` method, why not just collapse it as part of the parsing of the proto? Then you don't need this type at all and just use idiomatic `PathBuf`s everywhere.

Reply:
Really, this approach seems better, [fixed](https://github.com/google/rrg/pull/34/commits/59a5e28575045d37b15281d808bcccc6569d1eaa).